state/state.go

```diff
@@ -663,14 +651,42 @@ func chartNameWithoutRepository(chart string) string {
 	return chartSplit[len(chartSplit)-1]
 }
 
-func (state *HelmState) flagsForRelease(helm helmexec.Interface, basePath string, release *ReleaseSpec) ([]string, error) {
+func (state *HelmState) flagsForUpgrade(helm helmexec.Interface, basePath string, release *ReleaseSpec) ([]string, error) {
 	flags := []string{}
+	if release.Verify {
+		flags = append(flags, "--verify")
+	}
+	if release.Wait {
+		flags = append(flags, "--wait")
+	}
 	if release.Version != "" {
 		flags = append(flags, "--version", release.Version)
 	}
-	if release.Verify {
```

Comment:
`verify: true` in helmfile.yaml should have been failing on `helmfile lint` and `helmfile diff` due to that the unsupported `--verify` flag is passed.

This PR also fix that.